src/plots/array_container_defaults.js

```diff
@@ -64,4 +67,13 @@ module.exports = function handleArrayContainerDefaults(parentObjIn, parentObjOut
 
         contOut.push(itemOut);
     }
+
+    // in case this array gets its defaults rebuilt independent of the whole layout,
+    // relink the private keys just for this array.
+    if(Lib.isArray(previousContOut)) {
+        var len = Math.min(previousContOut.length, contOut.length);
+        for(i = 0; i < len; i++) {
+            Lib.relinkPrivateKeys(contOut[i], previousContOut[i]);
+        }
+    }
```

Comment:
My benchmark for success is just whether it passes the component test that was failing due to the `._remove` property getting lost. It was originally just called `.remove` which would still get lost, so I'll check and submit that fix as a separate PR as needed.